tests/cephfs/kernel_update.py

```diff
@@ -13,30 +13,47 @@ def run(ceph_cluster, **kw):
     Update the kernel of client nodes
     Reboot the client nodes
     Verify kernel is updated
+    RHEL-8 package installation order:
+    kernel-core
+    kernel-modules
+    kernel
+    RHEL-9 package installation order:
+    kernel-modules-core
+    kernel-core
+    kernel-modules
+    kernel
     """
     try:
-        fs_util = FsUtils(ceph_cluster)
         clients = ceph_cluster.get_ceph_objects("client")
-
+        test_data = kw.get("test_data")
+        log.info("Test data is " + str(test_data))
+        url = test_data.get("custom-config")[0]
+        log.info("Given Kernel Information is " + str(url))
+        verification_type = []
+        if "https://s3.upshift.redhat.com/DH-PROD-CKI/internal" in url:
+            log.info("This is a pre-verification")
+            verification_type.append("pre")
+        else:
+            log.info("This is a Post verification")
+            verification_type.append("post")
         log.info("checking Pre-requisites")
         if not clients:
             log.info(
                 f"This test requires minimum 1 client nodes.This has only {len(clients)} clients"
             )
             return 1
-        repo_file = clients[0].remote_file(
-            sudo=True, file_name="/etc/yum.repos.d/rh_add_repo.repo", file_mode="r"
-        )
-        file_contents = repo_file.readlines()
-        url = file_contents[1]
+        ceph_nodes = ceph_cluster.get_nodes()
         kernel_package = url.split("/")[-1]
-
-        if "p" in file_contents[0]:
-            url = file_contents[2].split("=", 1)[1]
+        if "pre" in verification_type:
             log.info(url)
             kernel_package = url.split("/")[-1]
-            for client in clients:
-                kernel_version, _ = client.exec_command(sudo=True, cmd="uname -r")
+            for cnode in ceph_nodes:
+                cnode.exec_command(
+                    sudo=True,
+                    cmd=f"echo -e '[rh_add_repo]\nbaseurl={url}\ngpgcheck=0\nenabled=1\n"
```

Comment:
Is it possible to check if the URL exists before adding the repo ?

When manga002 is down, all upgrades will be skipped saying already upto date.

Reply:
@hkadam134 I think I can ping to magna002, but URL is not in magna002..?